### DISCUSSION TITLE
Using syntax to disambiguate python mode from subprocess mode.

### DISCUSSION THREAD

**Discussion by gtusr** [xonsh/xonsh#5183] [Ideas]
What about using Unicode characters to disambiguate syntax like "ls -l" instead of semantics (which in this instance would entail looking up definitions for 'ls' and 'l' as variables in the current session)?

For example, 'figure dash' or 'en dash' could be used to replace minus sign for specifying command options in subprocess mode.

**Comment by anki-code** [comment]
To remove disambiguating you can use explicit syntax e.g. `$[ls -l]`. Please read the [docs](https://xon.sh/tutorial.html#python-mode-vs-subprocess-mode) or [cheatsheet](https://github.com/anki-code/xonsh-cheatsheet/).

**Comment by gtusr** [comment]
Thanks for your reply.  I was already aware of that syntax, and on second thought I think that I prefer it to my original suggestion.
